### ISSUE TITLE
[642039] synergy: switch to another computer causes immediate gdm restart

### ISSUE THREAD

**Issue by epakai** [epakai/synergy-debian#12]

**Comment by epakai** [comment]
BTS_msg_id: <4E7655C6.30701@pobox.com>
BTS author: Moshe Yudkowsky <moshe@pobox.com>

Package: synergy
Version: 1.3.6-1
Severity: grave
Justification: renders package unusable

I have synergy server running on Linux, and a synergy client running on 
the Mac. Both client and server run at 1.3.6.

When I move my cursor from the Linux to the Mac, from time to time gdm 
will immediately halt and restart. This makes using syslog chancy at best.

I have a ~/synergy.conf file - note that it does *not* have a leading 
"." but works regardless. The contents are:

section: screens
	linux_computer:
	mac_computer:
end
section: links
	linux_computer:
			left(40,60) = mac_computer(40,60)
	mac_computer:
			right(40,60) = linux_computer(40,60)
end
section: options
	switchCorners = all
	#switchDelay = 500
end



-- System Information:
Debian Release: wheezy/sid
   APT prefers unstable
   APT policy: (990, 'unstable'), (500, 'testing'), (500, 'stable'), (1, 
'experimental')
Architecture: amd64 (x86_64)

Kernel: Linux 2.6.39-2-amd64 (SMP w/2 CPU cores)
Locale: LANG=en_US.utf8, LC_CTYPE=en_US.utf8 (charmap=UTF-8)
Shell: /bin/sh linked to /bin/bash

Versions of packages synergy depends on:
ii  libc6         2.13-21
ii  libgcc1       1:4.6.1-3
ii  libice6       2:1.0.7-2
ii  libsm6        2:1.2.0-2
ii  libstdc++6    4.6.1-3
ii  libx11-6      2:1.4.4-1
ii  libxext6      2:1.3.0-3
ii  libxinerama1  2:1.1.1-3
ii  libxtst6      2:1.2.0-3

synergy recommends no packages.

synergy suggests no packages.

-- no debconf information

-- 
Moshe Yudkowsky * moshe@pobox.com * www.pobox.com/~moshe
  "Critical thinking is what tells you when to say 'Oh really?'
   instead of 'Oh cool!'"
    			-- Barry Gehm

**Comment by epakai** [comment]
BTS_msg_id: <4E76590C.6010003@debian.org>
BTS author: Jeff Licquia <licquia@debian.org>

On 09/18/2011 04:34 PM, Moshe Yudkowsky wrote:
> I have synergy server running on Linux, and a synergy client running on
> the Mac. Both client and server run at 1.3.6.
>
> When I move my cursor from the Linux to the Mac, from time to time gdm
> will immediately halt and restart. This makes using syslog chancy at best.
>
> I have a ~/synergy.conf file - note that it does *not* have a leading
> "." but works regardless. The contents are:

I'm a little confused.  Are you running synergys on startup?  If so, it 
will not read your per-user synergy.conf (unless you're giving it the 
config file on the command line, in which case its name makes no 
difference).

If your user session is just crashing, giving you the gdm login window 
once the crash has settled down, please just say that.

I'm also unclear as to what the problem with syslog is.

Can you try a few things to help and see where the problem might be? 
Tell me if changing your configuration in the following ways helps:

  - Dropping the ranges in the links.

  - Dropping the "switchCorners" option.

**Comment by epakai** [comment]
BTS_msg_id: <4E76691A.3050201@pobox.com>
BTS author: Moshe Yudkowsky <moshe@pobox.com>

Thanks for writing.

Jeff Licquia wrote, On 09/18/2011 03:48 PM:

> I'm a little confused.  Are you running synergys on startup?  If so, it 
> will not read your per-user synergy.conf (unless you're giving it the 
> config file on the command line, in which case its name makes no 
> difference).

I'm running synergys from the command line after I log into gdm.

> If your user session is just crashing, giving you the gdm login window 
> once the crash has settled down, please just say that.

The user session crashes and I get the login window once the crash has 
settled down.

> I'm also unclear as to what the problem with syslog is.

Sorry, that was a "braino," an error similar to a typo but caused by 
staying up until 1 AM sorting out issues... I meant to say that 
_synergys_ caused gdm restarts and made using the otherwise-excellent 
and vital synergys very chancy.

> Can you try a few things to help and see where the problem might be? 
> Tell me if changing your configuration in the following ways helps:
> 
>  - Dropping the ranges in the links.
> 
>  - Dropping the "switchCorners" option.

I've dropped back to synergy/stable (1.3.1).

* I'll report back if it synergys still fails
* If it still fails, I will remove "switchCorners" and then afterwards 
the ranges and report back.

I have not measured the uptime, but with 1.3.6 (unstable) I'd generate a 
failure every few hours with moderate usage. I will report back when I 
have more data.

If I do not generate any failures with 1.3.1, I will test in the 
evenings when I am not developing. Each crash takes some time to recover 
from -- Gnome does not restore windows to workspaces for reasons I've 
never figured out -- including possible lost data.


-- 
Moshe Yudkowsky * moshe@pobox.com * www.pobox.com/~moshe
  "It's very windy today... watch out for flying phone booths
   on the East-West Tollway."
    			-- Chicago traffic report

**Comment by epakai** [comment]
BTS_msg_id: <4E769177.6060904@pobox.com>
BTS author: Moshe Yudkowsky <moshe@pobox.com>

I've just had another gdm crash (screen blanks, login screen appears) as 
I attempted to switch screens. synergy 1.3.1 was up for 4 hours, 20 
minutes before crashing.

I've restarted 1.3.1 without the switchCorners option:

> section: options
>         #switchCorners = all
>         #switchDelay = 500
> end

I'll report back when I have either another crash or substantial usage 
without a crash.


-- 
Moshe Yudkowsky * moshe@pobox.com * www.pobox.com/~moshe

**Comment by epakai** [comment]
BTS_msg_id: <4E775B59.60308@pobox.com>
BTS author: Moshe Yudkowsky <moshe@pobox.com>

Synergy version 1.3.1 crashed after about 5 hours of use. The conf file 
   still had the "links" 40-60% as above but with switchCorners still 
commented out as above.

I've downloaded 1.3.7 from the synergy site to see if that works. If 
that does not work, I'll try removing the links - that will be awkward 
to work with but I'll have little choice.

-- 
Moshe Yudkowsky * moshe@pobox.com * www.pobox.com/~moshe

**Comment by epakai** [comment]
BTS_msg_id: <4E779FAE.9050000@pobox.com>
BTS author: Moshe Yudkowsky <moshe@pobox.com>

I should mention that in each case when there was a crash, I moved the 
mouse very rapidly from one screen to the other.

-- 
Moshe Yudkowsky * moshe@pobox.com * www.pobox.com/~moshe

**Comment by epakai** [comment]
BTS_msg_id: <4E77B71B.8020908@pobox.com>
BTS author: Moshe Yudkowsky <moshe@pobox.com>

Deb from the Synergy web site, 1.3.7, crashed after about 7 hours, a new 
record. synergy.conf as above (no switchCorners, yes limits).

-- 
Moshe Yudkowsky * moshe@pobox.com * www.pobox.com/~moshe

**Comment by epakai** [comment]
BTS_msg_id: <4E77BC95.7060801@debian.org>
BTS author: Jeff Licquia <licquia@debian.org>

On 09/19/2011 04:01 PM, Moshe Yudkowsky wrote:
> I should mention that in each case when there was a crash, I moved the
> mouse very rapidly from one screen to the other.

Thanks for all the testing; it's very helpful.

When you move the mouse rapidly between screens, does it crash every 
time?  If not, does it seem to make crashes happen more quickly?

**Comment by epakai** [comment]
BTS_msg_id: <4E77C786.5050100@pobox.com>
BTS author: Moshe Yudkowsky <moshe@pobox.com>

Jeff Licquia wrote, On 09/19/2011 05:05 PM:
> On 09/19/2011 04:01 PM, Moshe Yudkowsky wrote:
>> I should mention that in each case when there was a crash, I moved the
>> mouse very rapidly from one screen to the other.
> 
> Thanks for all the testing; it's very helpful.
> 
> When you move the mouse rapidly between screens, does it crash every 
> time?  If not, does it seem to make crashes happen more quickly?

Rapid movement does _not_ cause a problem each time, but it seems that 
each time there is a problem, it was during rapid movement.

Right now I'm testing the Synergy 1.4.x deb - I'm trying hard to avoid 
removing the limits because of my setup here.

It suddenly occurs to me that my setup here may be an additional issue 
-- conflict over what the edge hit means. I have GNOME on top of 
elightenment window manager, with multiple workspaces/desktops (3 
workspaces in a 3x1 array, each workspace contains a 2x2 array of 
virtual desktops). When inside a workspace I like to mouse between each 
set of 2x2 desktops and I'd wouldn't have access to my left windows if I 
were to remove the limits - any movement to the left would push me onto 
the other screen.

So the issue might be contention between enlightenment and synergy over 
the meaning of the mouse hit at the left edge...?

I'd switch to a different window manager if I could find one that 
switches between virtual desktops when I hit the edge of the current 
desktop. In fact no others seem to allow virtual desktops inside a 
workspace, or perhaps I'm reading the documentation incorrectly.

Thanks for your patience. I'll report back when I have further details.




-- 
Moshe Yudkowsky * moshe@pobox.com * www.pobox.com/~moshe
  "Don't blame the rainbows for the rain."
    					-- The Moody Blues

**Comment by epakai** [comment]
BTS_msg_id: <4E77EB60.8030208@pobox.com>
BTS author: Moshe Yudkowsky <moshe@pobox.com>

1.4.4 crashed after about 4 hours. I am now running Debian unstable, 1.3.6.

I'm trying switchDelay=500 to see if that helps slow things down. I 
don't see any visible effect from using the option when I test it -- 
that is, I don't see any delay in switching between screens.


-- 
Moshe Yudkowsky * moshe@pobox.com * www.pobox.com/~moshe

**Comment by epakai** [comment]
BTS_msg_id: <4E77EDF9.10103@debian.org>
BTS author: Jeff Licquia <licquia@debian.org>

On 09/19/2011 06:51 PM, Moshe Yudkowsky wrote:
> I'd switch to a different window manager if I could find one that
> switches between virtual desktops when I hit the edge of the current
> desktop. In fact no others seem to allow virtual desktops inside a
> workspace, or perhaps I'm reading the documentation incorrectly.

I know it would be painful to your current mode of working, but it might 
also help if you could run a "standard" GNOME desktop, with the default 
workspace layout and no edge limits on synergy.  If you got through, 
say, a whole day without crashing, that might tell us something.

I should have mentioned this before, too, but could you run synergys 
with "-d DEBUG1 -l <logfile>" and send me the resulting log once it 
crashes again?

**Comment by epakai** [comment]
BTS_msg_id: <4E77EFAC.9060803@pobox.com>
BTS author: Moshe Yudkowsky <moshe@pobox.com>

Jeff Licquia wrote, On 09/19/2011 08:35 PM:

> I should have mentioned this before, too, but could you run synergys 
> with "-d DEBUG1 -l <logfile>" and send me the resulting log once it 
> crashes again?

In progress.

> I know it would be painful to your current mode of working, but it might also help if you could run a "standard" GNOME desktop, with the default workspace layout and no edge limits on synergy.  If you got through, say, a whole day without crashing, that might tell us something

I'm desperate enough to do that next.



-- 
Moshe Yudkowsky * moshe@pobox.com * www.pobox.com/~moshe
"So many of them, and so few of us, how are we ever going to bury them all?"
Robert Frezza, _Fire in a Faraway Place_

**Comment by epakai** [comment]
BTS_msg_id: <4E77F6E1.1040102@pobox.com>
BTS author: Moshe Yudkowsky <moshe@pobox.com>

I should also mention that the problem only seems to happen as I move 
from right (Linux server) to left (MacOS client).

-- 
Moshe Yudkowsky * moshe@pobox.com * www.pobox.com/~moshe
   signature v1.0

**Comment by epakai** [comment]
BTS_msg_id: <4E785A52.1010308@pobox.com>
BTS author: Moshe Yudkowsky <moshe@pobox.com>

This morning both machines were in screensaver mode. Moving the mouse 
resulted in an immediate crash - I don't know what machine "had" the mouse.

Log to follow direct to Jeff.


-- 
Moshe Yudkowsky * moshe@pobox.com * www.pobox.com/~moshe
  "Mother, can't you hear the hissing/Of the bullets all too plain.
   I may be numbered with the missing,/But never, never with the slain."
     			-- "Just Behind the Battle, Mother"

**Comment by epakai** [comment]
BTS_msg_id: <4E78675F.10801@pobox.com>
BTS author: Moshe Yudkowsky <moshe@pobox.com>

When looking through the various logs, I noticed that at each crash the 
entry "gmd_slave_xioerror_handler" and a notation about fatal and 
restart appeared in the logs (daemon.log, syslog, etc.).

That's a generic error message AFAICT from searching the web, but it got 
me to thinking: I'd had a crash earlier when I plugged in my 
xsane-supported scanner. I'd put it off as unrelated... I then wondered 
if it were USB related, (if the USB hotplug of a the scanner caused a 
crash, could the USB keyboard/mouse cause a crash?), but USB hotplugs of 
flash drives did not crash gdm.

Reading through the web I decided to try downgrading to 
nvidia-kernel-dkms/testing.

This has fixed (a) my problem with the scanner and (b) fixed my problem 
with bug #641991. I'm going to continue working with this version of 
nvidia and report back. I'm at synergy 1.3.6 at the present time.

-- 
Moshe Yudkowsky * moshe@pobox.com * www.pobox.com/~moshe
  "Thanks to radio, TV, and the press we can now develop absurd
   misconceptions about peoples and governments we once hardly knew
   existed."	-- Charles Fair, _From the Jaws of Victory_

**Comment by epakai** [comment]
BTS_msg_id: <4E7899CF.5060108@pobox.com>
BTS author: Moshe Yudkowsky <moshe@pobox.com>

I've just spent a couple of hours attempting to get nvidia to work; I've 
given up and gone back to unstable as my nvidia-kernel-dkms/unstable.

So: I withdraw my prior remarks about nvidia. I did get a gdm crash and 
I don't know why. I'm going to try a 1000ms switch delay, maybe that 
will work.


-- 
Moshe Yudkowsky * moshe@pobox.com * www.pobox.com/~moshe
  "Oh, they couldn't hit an elephant at this dist..."
   General John Sedgwick, 9 May 1864

**Comment by epakai** [comment]
BTS_msg_id: <4E79E52B.5050702@pobox.com>
BTS author: Moshe Yudkowsky <moshe@pobox.com>

I found that 1.3.6 (current unstable) crashes in a few short minutes of 
usage,even if I remove the "limits" that restrict the window for movement.

At present I'm trying 1.3.7 from the synergy website, which is their 
"stable," and I'm using no limits as well as 100ms delays at the borders.

-- 
Moshe Yudkowsky * moshe@pobox.com * www.pobox.com/~moshe
  "Every morning you will see the wolf lying on the pasture side by
   side with a sheep. The only problem is that every morning it will
   be a different sheep."		-- Mohammed Wattad, M.K.

**Comment by epakai** [comment]
BTS_msg_id: <4E7FF481.8000709@debian.org>
BTS author: Jeff Licquia <licquia@debian.org>

severity 642039 normal
thanks

We've transmitted some logs.  At this point, I'm not convinced that 
there isn't some issue with the X server or video driver here, and I 
can't reproduce the problem.  So I'm dropping the severity to normal.

**Comment by epakai** [comment]
BTS_msg_id: <4E88F47A.2070800@debian.org>
BTS author: Jeff Licquia <licquia@debian.org>

I gave Moshe a few days to make sure his new-found stability wasn't a 
fluke; since there's been no news since then, I think we can conclude 
the problem was xorg-related, and not synergy-related.  Please reopen or 
file a new bug if this comes back.

-------- Original Message --------
Subject: working after xorg update
Date: Tue, 27 Sep 2011 05:13:10 -0500
From: Moshe Yudkowsky <moshe@pobox.com>
Organization: The Institute
To: Jeff Licquia <licquia@debian.org>

My latest update, which included xorg updates, seems to have fixed some
minor and not-so-minor annoyances (xsane no longer crashes Gnome).
Synergy 1.4.4 worked well for a day but seemed a bit slow at times,
reason unknown; I've just loaded 1.3.7 from the website's deb and we'll
see if that works just as well.

In other words, at this point I believe the problem was fixed by xorg
updates.

If something interesting comes up I'll be in touch.
-- 
Moshe Yudkowsky * moshe@pobox.com * www.pobox.com/~moshe
"There was an crash on the expressway this morning. Fortunately, all the
traffic was backed up."
    				-- Eli Yudkowsky

**Comment by epakai** [comment]
BTS_msg_id: Bug archived.
BTS author: Debbugs Internal Request <owner@bugs.debian.org>

# A New Hope
# A long time ago, in a galaxy far, far away
# something happened.
#
# Magically this resulted in the following
# action being taken, but this fake control
# message doesn't tell you why it happened
#
# The action:
# Bug archived.
thanks
# This fakemail brought to you by your local debbugs
# administrator
